### PR TITLE
writing test for calibrate with data mapped to observables

### LINKED ISSUE #439: Calibrate does not seem to work when data is mapped to observables
Calibrate works fine when columns in a given dataset are mapped to state variables (like "I" or "H"), but fails when data columns are mapped to observables (like "infected" or "hospitalized"). Please see https://github.com/ciemss/pyciemss/blob/sa-load-data-tests/docs/source/sa-testing-notebooks/tests_for_load_data.ipynb for example.

### PR DESCRIPTION
**Pull request by sabinala**
This PR adds a test to make sure that `calibrate` works when data is mapped to observables, rather than state variables (in which case the test should be skipped).

Closes #439 
(Note that this doesn't solve the issue, just creates a test to address it! See Issue #447 for the fix)

### CODE REVIEW

**Comment by sabinala** [comment]
@SamWitty sorry, I should have just messaged you instead of requesting your review! My intention was to ask that you take a look at this and let me know what I can do to fix it so that the tests pass.

File: tests/test_interfaces.py [hunk 1]
```diff
@@ -28,7 +28,10 @@ def dummy_ensemble_sample(model_path_or_json, *args, **kwargs):
 
 
 def setup_calibrate(model_fixture, start_time, end_time, logging_step_size):
-    if model_fixture.data_path is None:
+    if (
+        model_fixture.data_path is None
+        or model_fixture.data_mapped_to_observable == True
```
Comment (SamWitty) [inline tests/test_interfaces.py:33]: Remove == True should fix the linting.

**Comment by SamWitty** [comment]
@sabinala , with my changes to the linter is this now ready for review?

**Comment by sabinala** [comment]
@SamWitty yes, and thank you for making those changes!

File: tests/test_interfaces.py [hunk 2]
```diff
@@ -423,6 +423,36 @@ def test_output_format(
     assert processed_result["sample_id"].dtype == np.int64
 
 
+# Failing test for calibration when data is mapped to observables
+@pytest.mark.skip("Failing test for calibration when data is mapped to observables")
+@pytest.mark.parametrize("model_fixture", MODELS)
+@pytest.mark.parametrize("start_time", START_TIMES)
+@pytest.mark.parametrize("end_time", END_TIMES)
+@pytest.mark.parametrize("logging_step_size", LOGGING_STEP_SIZES)
+def test_calibrate_with_observables(
```
Comment (SamWitty) [inline tests/test_interfaces.py:432]: I'm not sure that this test is necessary, as the existing `test_calibrate` will already create a test and then skip it based on the new changed to `setup_calibrate`.

Comment (SamWitty) [reply tests/test_interfaces.py:432]: My suggestion is to remove this function, and then when we fix the bug we simple remove your addition on line 32 of this file.

**Review by SamWitty — CHANGES_REQUESTED** [review]
@sabinala, looks good, but this can be made simpler and reduce redundant tests. See comments.

**Comment by sabinala** [comment]
@SamWitty I removed the test, but kept everything else.

File: tests/fixtures.py [hunk 3]
```diff
@@ -8,6 +8,7 @@
 
 MODELS_PATH = "https://raw.githubusercontent.com/DARPA-ASKEM/simulation-integration/main/data/models/"
 DATA_PATH = "https://raw.githubusercontent.com/DARPA-ASKEM/simulation-integration/main/data/datasets/"
+DATA_PATH2 = "https://raw.githubusercontent.com/ciemss/pyciemss/sa-load-data-tests/docs/source/sa-testing-notebooks/"
```
Comment (SamWitty) [inline tests/fixtures.py:11]: This path does not look stable, especially once the branch is deleted. Could you please push these data files to the existing `DATA_PATH` above.

**Review by SamWitty — CHANGES_REQUESTED** [review]
Apologies, I just noticed another minor issue that I didn't notice previously.

File: tests/fixtures.py (hunk 3 above)
Comment (sabinala) [reply tests/fixtures.py:14]: @SamWitty I just made a PR to add the dataset to the original `MODELS_PATH`

**Comment by sabinala** [comment]
@SamWitty once this PR (https://github.com/DARPA-ASKEM/simulation-integration/pull/68) is merged, this PR #450 should be good to go

**Comment by SamWitty** [comment]
> @SamWitty once this PR ([DARPA-ASKEM/simulation-integration#68](https://github.com/DARPA-ASKEM/simulation-integration/pull/68)) is merged, this PR #450 should be good to go

Turns out that I have permissions to approve PRs on `simulation-integration`. Who knew!

**Review by SamWitty — APPROVED** [review]

**Comment by sabinala** [comment]
@SamWitty oh noooooooo! One of these tests isn't passing. It looks like an error from `dummy_ensemble_sample` but I'm not sure what's happening. What do you think? Does this look like a problem with the model?

**Comment by SamWitty** [comment]
> @SamWitty oh noooooooo! One of these tests isn't passing. It looks like an error from `dummy_ensemble_sample` but I'm not sure what's happening. What do you think? Does this look like a problem with the model?

There's a lot of randomness in the tests, and sometimes they fail. It's very challenging to test algorithms that are fundamentally random. I just rerun failing tests.

**Pull request merged**